src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -103,6 +105,9 @@ public function warmUp($cacheDir)
 
     public function addResource($format, $resource, $locale, $domain = null)
     {
+        if ($this->resourceFiles) {
+            $this->addResourceFiles();
```

Comment:
do we need to do it here ?

We could ensure we do it only on `initialize` by calling `parent::addResource` for the resource files instead of `$this->addResource` (avoiding to copy them first from `resourceFiles` to `resources`, to add them in the parent later.

As `initialize` will process `resourceFiles` before `resources`, the other will be preserved.